src/pty.rs

```diff
@@ -52,15 +47,10 @@ pub fn ptsname_r(fd: RawFd) -> Result<String> {
         return Err(Error::last().into());
     }
 
-    let name_cstr = unsafe { CStr::from_ptr(name_buf.as_ptr()) };
-    let name = match name_cstr.to_str() {
-        Ok(s) => s.to_string(),
-        Err(_) => {
-            // FIXME: Unsure what error I should report here.
-            return Err(Error::last().into());
-        }
+    let name = unsafe {
+        CStr::from_ptr(name_buf.as_ptr())
     };
-    Ok(name)
+    Ok(name.to_string_lossy().into_owned())
```

Comment:
We should never receive invalid UTF-8 sequences from the OS, so that should never happen. I'd prefer to use `from_utf8_unchecked` (as you mentioned in your comment), which doesn't do any error detection, but I couldn't figure out how to use it. And note that using `from_utf8_unchecked` has the exact same problem, but instead of fixing invalid UTF-8 strings it will just crash later when it's attempted to be decoded as proper UTF-8.

Basically if we use the `String` type as the output from this function, it **must** be valid UTF-8 regardless of how we convert it. So the question really is, does `ptsname*()` **always** return valid UTF-8. I think the answer is yes, but I don't really know.